bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/NetworkUtils.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.broadlink.internal;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.ServerSocket;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.Enumeration;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+
+/**
+ * Utilities for working with the local network.
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+@NonNullByDefault
+public class NetworkUtils {
+
+    public static boolean hostAvailabilityCheck(@Nullable String host, int timeout, Logger logger) {
+        if (host == null) {
+            logger.warn("Can't check availability of a null host");
+            return false;
+        }
+        try {
+            InetAddress address = InetAddress.getByName(host);
+            return address.isReachable(timeout);
```

Comment:
This utility is used to determine / update the status of the thing. It is called by updateItemStatus() in the BroadlinkBaseThingHandler, to ensure we set the correct thing state.

Reply:
My question remains the same. This is an out of context check and therefore not so usefull. Let's sey there is a sequence of events:
1. the bindings inits or performans an update or other event that `updateItemStatus()` is called. 
2. Everytime `updateItemStatus()` is called the host is checked for reachability
3. The host may become unavailable between this check and the actuall API-call retrieving the data. Making it essential to also handle that exception at the actuall API-call. 

2 is overhead and 3 shows this check has no use as it is not safe.
Also see https://github.com/openhab/openhab-addons/pull/16273/files#r1659249350

Reply:
After looking at the situation, I think the binding is trying to do the following. When there is an event or on a regular basis the uipdateItemStatus() method is called. It first checks for the reachability of the thing. If the host is not reachable interesting things happen. If the IP is defined as static it does nothing. However, if it is dynamic it starts a thread to discover if the device is on a new IP address. I think this is the primary reason for the reachability check. Ideally, this should be rewritten to a more clean design. However, this is complicated as it breaks the structure of the current implementation due to inheritance relationships.

To be honest, for me this is not a show stopper and rather a nice to have then a must have. The main reason is that in most cases the overhead is not significant.

Reply:
The mechanism is not wrong, it is the implementation that is far from optimal. AS stated above, there are unhandlked race conditions and it is called excessive.
It can be fixed by adjusting it a little bit. 
Should not be that much work. I allready created an example pseudo code  here https://github.com/openhab/openhab-addons/pull/16273/files/1cbc3968e5e955e21844f35431a1aaec7f7a33b5#r1659249350

that can be extended with a catch block specific for when the host is not reachable: (have to check the right thrown exception classes, used it out of my head) 
```
try {
    getStatusFromDevice();
    updateStatus(ThingStatus.ONLINE);
} catch (NoRouteToHostEception | HostNotFoundException ex) {
    // this block should be hit whenever the host is not reachable. Here you can run recovery code:
            if (thingConfig.isStaticIp()) {
                    forceOffline(ThingStatusDetail.NONE, "Couldn't find statically-IP-addressed device");
            } else {
                DeviceRediscoveryAgent dra = new DeviceRediscoveryAgent(thingConfig, this);
                dra.attemptRediscovery();
                logger.debug("Asynchronous dynamic IP device search initiated...");
            }
    }
} catch (CustomAuthenticationException ex) {
    //update Thing state to OFFLINE set detail to CONFIGURATION_ERROR nad message to "Credentials are invalid";
    }
} catch (CustomCommuncationException ex) {
    //update Thing state to OFFLINE set detail to COMMUNICATION_ERROR and message to "Fatal timeout after 3 retries"
    }
}
```

Reply:
The code has been updated.